rinex/src/algorithm/averaging.rs

```diff
@@ -1,58 +1,43 @@
-use crate::prelude::*;
+use hifitime::{Duration, Epoch};
 
-pub enum AverageType {
-    MovingAverage,
-    Exponential,
-    Cummulative,
+fn moving_average<T: std::default::Default>(
+    data: Vec<(Epoch, T)>,
+    window: Duration,
+) -> Vec<(Epoch, T)> {
+    let mut acc = T::default();
+    let mut prev_epoch: Option<Epoch> = None;
+    let mut ret: Vec<(Epoch, T)> = Vec::new();
+    for (epoch, value) in data {}
+    ret
```

Comment:
Preliminary work, to be concluded soon. 
This will first be used in QC::clock drift analysis.  
But i'm also interested in providing the ability to smooth some datasets from the command line.
We can already do that but only on pseudo range observations, with a specific "Hatch" filter. 
Smoothing could apply to all datasets